nidx/nidx_vector/src/hnsw/build.rs

```diff
@@ -134,33 +141,55 @@ impl<'a, DR: DataRetriever> HnswBuilder<'a, DR> {
         result
     }
 
-    pub fn insert(&mut self, x: VectorAddr, hnsw: &mut RAMHnsw) {
-        match hnsw.entry_point {
-            None => {
-                let top_level = self.get_random_layer();
-                hnsw.increase_layers_with(x, top_level).update_entry_point();
+    /// Insert a node into all corresponding layers.
+    /// The node must be created first by calling initialize_graph()
+    pub fn insert(&mut self, node: VectorAddr, hnsw: &mut RAMHnsw) {
```

Comment:
NIT: I feel that initialize_graph and insert are not the most accurate names. Here insert is connecting a node with neighbours more than inserting in the graph, that is done in the initialize.